### ISSUE TITLE
Custom input prompts

### ISSUE THREAD

**Issue by pyroscope** [pyroscope/rtorrent-ps#49]
* Patch or copy the ``DownloadList::receive_view_input`` code
* Allow settings a custom ``title`` (prompt)
* Call specified command afterwards, with the input as arg.0
* ``ui.prompt=title,((command, foo, (argument.0)))`` or similar

**Comment by chros73** [comment]
I'm not sure whether it's needed. (I was asked about the same thing and also the other one (Command: ui.notice=duration,message) during the last week.)
E.g. I only use 1 parameterized command all the time `d.move_to=foo,1,0` that can have loads of possibilities. If somebody needs a feature like this means to me that he/she doesn't know how to use rtorrent :)

Instead of this what we definitely need is a history for the command prompt like in `bash` (I think it can solve this issue as well in a more efficient way):
- save the history file in session dir or in home dir if session isn't set
- set the max size of lines in it via a config variable
- use up/down arrows at command prompt to go through it
- search functionality can be a bonus but not a must have

**Comment by chros73** [comment]
Hmmm ... :) It's easy when you use [readline lib](http://cc.byexamples.com/2008/06/13/gnu-readline-how-to-keep-a-history-list-of-entered-command-lines/) but maybe it's an overkill for rtorrent.

**Comment by pyroscope** [comment]
No, readline is the standard implementation for this. And yes, that would probably serve the same purpose.

**Comment by chros73** [comment]
I've implemented the [input history](https://github.com/chros73/rtorrent-ps/issues/83) (I didn't use readline since every bit of the prompts was coded in rtorrent).
